src/chains/ethereum/ethereum/tests/api/eth/getTransactionReceipt.test.ts

```diff
@@ -0,0 +1,74 @@
+import getProvider from "../../helpers/getProvider";
+import assert from "assert";
+import EthereumProvider from "../../../src/provider";
+
+describe("api", () => {
+  describe("eth", () => {
+    describe("getTransactionReceipt", () => {
+      let provider: EthereumProvider;
+      let logger;
+
+      beforeEach(async () => {
+        // create a logger to test output
+        logger = {
+          clearLoggedStuff: function () {
+            this.loggedStuff = "";
+          },
+          loggedStuff: "",
+          log: function (message) {
+            if (message) {
+              this.loggedStuff = this.loggedStuff.concat(message);
+            }
+          }
+        };
+        provider = await getProvider({ logging: { logger } });
+      });
+
+      it("returns the receipt for the transaction", async () => {
+        const [from] = await provider.send("eth_accounts");
+        await provider.send("eth_subscribe", ["newHeads"]);
+
+        const hash = await provider.send("eth_sendTransaction", [
+          {
+            from,
+            to: from
+          }
+        ]);
+
+        // wait for the tx to be mined
+        await provider.once("message");
+        const receipt = await provider.send("eth_getTransactionReceipt", [
+          hash
+        ]);
+        assert(receipt);
+      });
+
+      it("returns null if the transaction does not exist", async () => {
+        const result = await provider.send("eth_getTransactionReceipt", [
+          "0x0"
+        ]);
+        assert.strictEqual(result, null);
+      });
+
+      it("logs a warning if the transaction hasn't been mined yet", async () => {
+        logger.clearLoggedStuff();
```

Comment:
I made a suggestion above to add an `afterEach` that does this for us. If you accept the suggestion you should accept this one too!

```suggestion
```